src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_TypeChar.cs

```diff
@@ -267,7 +267,8 @@ private Document GetDocument()
         private CompletionHelper GetCompletionHelper()
         {
             var document = GetDocument();
-            return CompletionHelper.GetHelper(document);
+            var service = document.Project.Solution.Workspace.Services.GetService<ICompletionHelperService>();
```

Comment:
you've repeated this like 10 times.  Why not just have CmpletionHelper.GetHelper(Document) do exactly this?